sky/packages/sky/lib/src/widgets/snack_bar.dart

```diff
@@ -38,7 +38,7 @@ class SnackBarAction extends Component {
   }
 }
 
-class SnackBar extends Component {
+class SnackBar extends StatefulComponent {
```

Comment:
I think requiring the consumer to specify the AnimationPerformance is a step backwards, and this class is a good example why:
* The consumer class must be stateful, requiring it to implement initState and syncConstructorArguments.
* The consumer class needs to manually control the performance.
* There's more boilerplate for the common case of being notified on completion/dismissal.

My goal with the Transition classes was to handle the common cases of animations and get rid of the boilerplate involved with dealing with an AnimationPerformance. I'd hate to get rid of that. Maybe we could have 2 modes for a Transition - either handling the performance for you, or the more flexible version where you own and control the performance?